src/decoder.js

```diff
@@ -90,17 +95,22 @@ export default class Decoder {
   decodeMethod(inputData) {
     const errorObject = { name: 'UNDECODED', params: [{ name: 'rawData', value: inputData, type: 'data' }] };
     if (inputData === '0x') return '';
+
     const methodID = inputData.slice(2, 10);
     const abiItem = this.methodIDs[methodID];
     if (abiItem) {
       const params = abiItem.inputs.map(item => item.type);
+
       const decoded = SolidityCoder.decodeParams(params, inputData.slice(10));
       return {
         name: abiItem.name,
         params: decoded.map((param, index) => {
           let parsedParam = param;
           if (abiItem.inputs[index].type.indexOf('uint') !== -1) {
-            parsedParam = web3.toBigNumber(param).toString();
+            parsedParam = Array.isArray(param) ?
+              param.map(singleNumber =>
+                web3Utils.toBigNumber(singleNumber).toString()) :
+              web3Utils.toBigNumber(param).toString();
```

Comment:
This process is duplicated, I propose to make it in separate function.

Reply:
good idea